src/main/java/org/folio/dataimport/handlers/actions/CreateInvoiceEventHandler.java

```diff
@@ -253,10 +259,7 @@ private long determineInvoiceLinesQuantity(ParsedRecord parsedRecord) {
       linesChunk.add(getLinePair(entry, requestContext));
 
       if (index % MAX_PARALLEL_SEARCHES == 0 || index == refNumberList.size()) {
-        CompletableFuture<List<Pair<Integer, PoLine>>> chunk = collectResultsOnSuccess(linesChunk);
-        if (chunk.isDone() && !chunk.isCompletedExceptionally()) {
-          response.addAll(chunk.join());
-        }
+        response.addAll(collectResultsOnSuccess(linesChunk).join());
```

Comment:
thread will be stuck on join() and will not be unblocked because it waits for the future that it should complete itself

Reply:
removed this update from this pr as for it is out of scope of original issue